### PR TITLE
refactor: surface up runtime error when adding view

### PR DESCRIPTION
**Pull request by haishengwu-okta**
okta-123

## PR Checklist

- [ ] Have you verified the basic functionality for this change?

- [ ] Added unit tests?

- [ ] Added e2e tests

- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?

### Issue:

- [OKTA-XXXXXX](https://oktainc.atlassian.net/browse/OKTA-XXXXXX)

### CODE REVIEW

File: src/v2/controllers/FormController.js [hunk 1]
```diff
@@ -40,11 +40,18 @@ export default Controller.extend({
...
+        }
+      }).last();
+    } catch (error) {
+      // This is the place where runtime error (NPE) happens at most of time.
+      // It has been swallowed by Q.js hence add try/catch to surface up errors.
+      this.options.settings.callGlobalError(error);
+      return;
+    }
 
     this.listenTo(this.formView, 'save', this.handleFormSave);
```
Comment (anipendakur-okta) [inline src/v2/controllers/FormController.js:56]: What if we moved this to line 49 inside `try` and remove the `return` in catch block?

Comment (haishengwu-okta) [reply src/v2/controllers/FormController.js:56]: that also works. and also don't really like add `return`
but feels like it's slightly better that being more explicit at which line cause trouble? :D

Comment (anipendakur-okta) [reply src/v2/controllers/FormController.js:56]: Agreed, I'm good either way. Also, if we move listenTo statement inside, it's a free catch block for anything that can possibly go wrong in it :) Probably personal preference plays a big role here..

Comment (haishengwu-okta) [reply src/v2/controllers/FormController.js:56]: also It's subject to change cause there may be better place to add `try/catch`

File: src/v2/controllers/FormController.js [hunk 2]
```diff
@@ -40,11 +40,18 @@ export default Controller.extend({
...
+    try {
+      this.formView = this.add(TheView, {
+        options: {
+          currentViewState,
+        }
+      }).last();
+    } catch (error) {
+      // This is the place where runtime error (NPE) happens at most of time.
+      // It has been swallowed by Q.js hence add try/catch to surface up errors.
+      this.options.settings.callGlobalError(error);
```
Comment (pradeepdewda-okta) [inline src/v2/controllers/FormController.js:52]: where is this fn callGlobalError defined ?

Comment (haishengwu-okta) [reply src/v2/controllers/FormController.js:52]: `src/models/Settings.js`

**Pull request merged**